Clean/Circuit/Basic.lean

```diff
@@ -284,6 +284,12 @@ def lookup (l: Lookup F) : Circuit F Unit :=
 
 end Circuit
 
+@[circuit_norm]
+def ProvableType.witness {α: TypeMap} [ProvableType α] (compute : Environment F → α F) : Circuit F (α (Expression F)) :=
+  modifyGet fun ops =>
+    let var := var_from_offset α ops.offset
+    ⟨var, .witness ops (size α) (fun env => compute env |> to_elements)⟩
```

Comment:
I moved this here from `Extensions`, I think it's generally useful, and better if defined using `modifyGet` and returns `var_from_offset`, since we already have good simp lemmas about that